### PR TITLE
Port blockquote block to RN mobile

### PR DESCRIPTION
**Pull request by SergioEstevao**
## Description

Implements the quote block in react native.

This is done with minimal changes on the quote block edit code. 
It implements a new Blockquote component that in the web continues to wrap the richtext using a blockquote element. On the native side it uses a View to implement the necessary styling and wrapping of the RichText component.

It also changes the RichText in native to a have a default tagName of `div` this allows the RichText component using inside the quote block to not to be changed and allow in native the proper wrapping of internal elements and final removal of the root tag.

~~This PR implement a specific native version of the edit of the block because of the following:~~

~~- Use of an HTML element, `<blockquote>` to set the style of the richtext~~
~~- Use of two richtext elements but not define the top element of each of them, so this need to be changed for mobile in order for their style to be presented correctly.~~

This can be tested using the following PR on GB-mobile:https://github.com/wordpress-mobile/gutenberg-mobile/pull/963

### CODE REVIEW

**Comment by hypest** [comment]
This seems to work nicely @SergioEstevao , nice one!

By the way, I think we might have an opportunity here to make this block cross-platform. I've pushed https://github.com/WordPress/gutenberg/commits/rnmobile/blockquote_port_to_mobile-cross-platform with some early experiments towards that. Have a look and we can chat about this!

File: packages/block-library/src/quote/blockquote.native.js
Comment (koke) [inline packages/block-library/src/quote/blockquote.native.js]: I think this can be simpler, Views in react native seem to support border styles:

```jsx
<View style={ styles.wpBlockQuote } >
	{ props.children }
</View>
```

```css
.wpBlockQuote {
	border-left-width: 4px;
	border-left-color: $black;
	border-left-style: solid;
	padding-left: 8px;
	margin-left: 8px;
}
```

I'll let Thomas review the exact colors, but the web version seems to be using black. If you're seeing a blue line, that's coming from Twenty Nineteen

File: packages/block-editor/src/components/rich-text/index.native.js
Comment (SergioEstevao) [inline packages/block-editor/src/components/rich-text/index.native.js]: This is replicating the behaviour of the web where there the default tagName is a Div.

This allows the blockquote block to work properly because the <p> are surrounded by the div.

Our other blocks should work correctly because they define a tagName and don't use the default.

Comment (hypest) [reply packages/block-editor/src/components/rich-text/index.native.js]: > are surrounded by the div

Hmm, how do the `div`s play a role in the native mobile case, considering that this change is in the `.native.js` file? 🤔

Comment (SergioEstevao) [reply packages/block-editor/src/components/rich-text/index.native.js]: If you look at the file we expect a tagName in several places to be defined in order to be able to produce the HTML. It also allows the wrapping of the multiple <p> that can exist in a blockquote inside a root tag.

Comment (hypest) [reply packages/block-editor/src/components/rich-text/index.native.js]: Oh, so it's for proper support of the `save` method of the block?

Let's add this expanded explanation to the PR description @SergioEstevao , cool?

File: packages/viewport/src/with-viewport-match.native.js
Comment (koke) [inline packages/viewport/src/with-viewport-match.native.js]: As mentioned on Slack, this isn't behaving like the web version. For now, let's make the AlignmentToolbar a fake component on native, and leave that and the viewport implementation for https://github.com/wordpress-mobile/gutenberg-mobile/issues/198

File: packages/viewport/src/index.native.js
Comment (koke) [inline packages/viewport/src/index.native.js]: I don't think we need this one anymore right?

File: packages/block-library/src/quote/edit.js
Comment (koke) [inline packages/block-library/src/quote/edit.js]: Following @gziolo's [comment on the other PR](https://github.com/WordPress/gutenberg/pull/15769#pullrequestreview-240492296), I'm not 100% sure about this naming.

I don't think this should be a standalone component, but having `Blockquote` inside of the `quote` block is a bit confusing to me. I think I'd suggest something along the lines of "Container" or "Wrapper" but I'd like to hear from @gziolo as well.

**Review by koke — COMMENTED** [review]
This looks pretty good for a first PR. Since the writing flow isn't ready to ship yet, make sure we only enable this for Debug builds, [like the other in progress blocks](https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/index.native.js#L23)

File: packages/block-library/src/quote/edit.js
Comment (gziolo) [reply packages/block-library/src/quote/edit.js]: I shared my thoughts already:

> Ideally, this new component would be renamed to `BlockQuotation` and placed inside `@wordpress/components` package (`packages/components/primitives/block-quotation.js`?) together with the corresponding documentation. See the prior art from @koke for `HorizontalRule` implemented in #14361.
> 
> Related MDN page which I used to reason about the name of the component: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/blockquote

Comment (gziolo) [reply packages/block-library/src/quote/edit.js]: You will have to run into the same issue when porting Pullquote block. It is also something that plugin developers will need as well.

File: packages/components/CHANGELOG.md
Comment (gziolo) [inline packages/components/CHANGELOG.md]: It might be 8.0.0 as well so it's safe to put and:
```suggestion
## Master
```
and update after release :)

**Review by gziolo — APPROVED** [review]
Thanks for addressing my comment. Refactoring for web part looks great. It should make porting Pullqote block much easier as it also uses `blockquote` HTML tag.

**Comment by SergioEstevao** [comment]
@koke do you want to give it another look, or can it be merged it in?

File: packages/block-library/src/quote/edit.js
Comment (youknowriad) [inline packages/block-library/src/quote/edit.js]: I wonder if we should gather all these Platform specific components in a separate package or otherwise under the same parent variable. 

Whether it's `import { Blockquote } from '@wordpress/platform'` 

or 

```js
import { PlatformComponents } from '@wordpress/components';
const { Blockquote } = PlatformComponents;
````

Comment (gziolo) [reply packages/block-library/src/quote/edit.js]: Yep, this is a good point. I think we should extract all those components into their own package and maybe re-export from `@wordpress/components` for backward compatibility only. It's definitely out of the scope of this PR @SergioEstevao. Anyway, it's something we should decide on quite soon as you will need to use more HTML tags like:
- `figure`
- `ul`, `ol`, `li`
- `img`
- `div`

**Comment by gziolo** [comment]
I see Travis detected some issue with style import:

> Error: File to import not found or unreadable: ./quote/style.scss.

Needs to be fixed to unblock merge.

**Comment by SergioEstevao** [comment]
I move the style file to be in the BlockQuotation package. It looks something was using it too.

File: packages/block-editor/src/components/rich-text/index.native.js
Comment (hypest) [inline packages/block-editor/src/components/rich-text/index.native.js]: This change (the addition of the "else" block with its setting of the `isSelected` flag) seems to be causing https://github.com/wordpress-mobile/gutenberg-mobile/issues/1031. @SergioEstevao , do you recall what this change was added to do? I'm going to open a PR to remove it but, I'm afraid I might be missing some context and the removal might add some other problem.

File: packages/components/src/primitives/block-quotation/style.scss
Comment (koke) [inline packages/components/src/primitives/block-quotation/style.scss]: This is the style for the block, I don't think it should have been moved to the component.

File: packages/block-library/src/quote/edit.js
Comment (koke) [reply packages/block-library/src/quote/edit.js]: What I meant is that, while this gets translated into a `blockquote` HTML element in the web version, I don't see it being a good standalone "quote" component right now, like the HorizontalRule is.

The current web implementation only looks like a quote because it's (IMO, wrongly) [stealing the styles for the quote block](https://github.com/WordPress/gutenberg/pull/15482/files#r287723792), which won't work when we want to do pullquote as well, and the native version is defining a non-configurable style as well.

Also, those styles are only applied to the `blockquote` because of the `wp-block-quote` class, but there's no specific mention of `blockquote` in the CSS. From what I see, all `blockquote` provides on the web is semantic markup, but nothing more over a generic `div`. Which is why, more abstractly, I wonder if it would make sense to make this a generic `View`/`Container` element, where one can pass a `tagName` prop for semantic reasons on the web implementation.

We still have the unresolved issue that we have to use inline styles for native, and CSS classes for the web, but otherwise the code could be pretty similar for both:

```jsx
<View tagName="blockquote" className={ className } style={ style }>
    <RichText
        identifier="value"
        // ...
    />
    { ( ! RichText.isEmpty( citation ) || isSelected ) && (
        <RichText
	    identifier="citation"
            // ...
	/>
    ) }
</View>
```

File: packages/components/src/primitives/block-quotation/style.scss
Comment (gziolo) [reply packages/components/src/primitives/block-quotation/style.scss]: Yes, you are correct. I missed it...

Comment (gziolo) [reply packages/components/src/primitives/block-quotation/style.scss]: This file was never referenced so it only needs to be deleted.

Comment (gziolo) [reply packages/components/src/primitives/block-quotation/style.scss]: https://github.com/WordPress/gutenberg/pull/15845 fixes it.

File: packages/block-editor/src/components/rich-text/index.native.js
Comment (hypest) [reply packages/block-editor/src/components/rich-text/index.native.js]: @SergioEstevao added a comment about this here: https://github.com/WordPress/gutenberg/pull/15833#issuecomment-496262819

**Pull request merged**
